### PR TITLE
Extended Stream for Strings and lower key map

### PR DESCRIPTION
**Pull request by jdotcms**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/util/LowerKeyMap.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package com.dotcms.util;
+
+import org.jetbrains.annotations.NotNull;
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/util/LowerKeyMap.java]: Nope.

File: dotCMS/src/main/java/com/dotcms/util/MapWrapper.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
+package com.dotcms.util;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+
+public class MapWrapper<K,V> implements Map<K,V> {
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/util/MapWrapper.java]: Needs to implement serializable

**Review by wezell — CHANGES_REQUESTED** [review]
What is this for?

**Review by nollymar — APPROVED** [review]

**Comment by nollymar** [comment]
@wezell I'm going to use it in a refactoring I'm doing in ImportUtil.java

File: dotCMS/src/main/java/com/dotcms/util/LowerKeyMap.java (hunk 1 above)
Comment (nollymar) [reply dotCMS/src/main/java/com/dotcms/util/LowerKeyMap.java]: Fixed. Now, I'm importing **com.dotcms.repackage.javax.validation.constraints.NotNull**

File: dotCMS/src/main/java/com/dotcms/util/MapWrapper.java (hunk 2 above)
Comment (nollymar) [reply dotCMS/src/main/java/com/dotcms/util/MapWrapper.java]: Done

File: dotCMS/src/main/java/com/dotcms/util/LowerKeyMap.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        this(new HashMap<>());
+    }
+
+    public LowerKeyMap(Map<String, V> delegate) {
+        super(delegate);
+    }
+
+    public LowerKeyMap(Map<String, V> delegate, Map<? extends String, ? extends V> baseMap) {
+        super(delegate);
+        this.putAll(baseMap);
```
[the remark this answers is not in the document]
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/util/LowerKeyMap.java]: make `putAll` final to overcome this?

**Review by dsilvam — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

**Pull request merged**